src/biguint.rs

```diff
@@ -434,6 +434,47 @@ impl One for BigUint {
 
 impl Unsigned for BigUint {}
 
+macro_rules! pow_impl {
+    ($unsigned:ty) => {
+         impl<'a> pow::Pow<$unsigned> for &'a BigUint {
+            type Output = BigUint;
+
+            #[inline]
+            fn pow(self, mut exp: $unsigned) -> Self::Output {
+                // square and multiply from num_traits::pow in the next branch
+                let mut base = self.clone();
+
+                if exp == 0 { return BigUint::one(); }
+
+                while exp & 1 == 0 {
+                    base = &base * &base;
+                    exp >>= 1;
+                }
+
+                if exp == 1 { return base; }
+
+                let mut acc = self.clone();
```

Comment:
Err, actually you do have a test for `pow(3)` -- but since that has no *trailing* zeros, `base` and `self` are still the same.  Try an exponent like 10 to see the failure I'm talking about.

Reply:
oops, I missed that when copying the pow implementation from `next`.

Reply:
Added a few tests as well.